### PR TITLE
feat(rule): add action matching completename

### PR DESCRIPTION
**Pull request by stonebuzz**
Add action to match on ```completename``` for :

- ```category```
- ```group assign```
- ```group observer```
- ```group requester```

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !23131

### CODE REVIEW

File: src/RuleCommonITILObject.php [hunk 1]
```diff
@@ -469,6 +511,20 @@ public function executeActions($output, $params, array $input = [])
                                     "name" => $regex_value,
                                 ]);
 
+                                // Stop at the first valid category found
+                                if ($result) {
+                                    $output['itilcategories_id'] = $category->getID();
+                                    break;
+                                }
+                            }
+                        } elseif ($field == "itilcategories_id_by_completename") {
```
Comment (cedric-anne) [inline src/RuleCommonITILObject.php:520]: It seems that naming convention is to prefix by a `_` when action does not match an existing field of the item.

```suggestion
                        } elseif ($field == "_itilcategories_id_by_completename") {
```

**Review by cedric-anne — APPROVED** [review]

File: src/RuleCommonITILObject.php (hunk 1 above)
Comment (stonebuzz) [reply src/RuleCommonITILObject.php]: you're right, fixed !

**Comment by trasher** [comment]
Tests are failing

File: src/RuleCriteria.php [hunk 2]
```diff
@@ -375,6 +375,11 @@ public function getRuleCriterias($rules_id)
      **/
     public static function match(RuleCriteria &$criterion, $field, &$criterias_results, &$regex_result)
     {
+        if (is_array($field)) {
+            $field = Sanitizer::decodeHtmlSpecialCharsRecursive($field);
+        } else {
+            $field = Sanitizer::decodeHtmlSpecialChars($field);
```
Comment (trasher) [inline src/RuleCriteria.php:381]: Maybe to fix tests:
```suggestion
            $field = Sanitizer::decodeHtmlSpecialChars($field ?? '');
```

**Review by trasher — APPROVED** [review]

**Pull request merged**